docs/developer/developer-manual.html

```diff
@@ -161,6 +161,24 @@ <h2 id="Build_tasks">Build tasks</h2>
 main directory or any subproject runs the <code>allTests</code> task only in the <code>framework</code> project.
 </p>
 
+<p>
+An iterative build-test-debug cycle where you assemble the Checker Framework
+and execute its tests can sometimes lead to a state where Gradle is not able to
+compile and build the system for no obvious reason.
+This is especially common when you kill tests before they fully finish, or kill
+compilation before it is fully complete, leading to a bad state or an
+inconsistent build cache.
```

Comment:
FWIW I haven't seen the inconsistent build cache issue before.  Generally, the Gradle build cache only gets updated after a task succeeds.  It could be a bug in our Gradle build logic somewhere...